jasmine/private/runner.js

```diff
@@ -1,4 +1,65 @@
 const path = require("path");
-const jasmine_path = require.resolve("jasmine");
 
-require(path.join(jasmine_path, "..", "..", "bin", "jasmine.js"));
+const shardCount = Number(process.env.TEST_TOTAL_SHARDS);
+const shardIndex = Number(process.env.TEST_SHARD_INDEX);
+const shardStatusFile = process.env.TEST_SHARD_STATUS_FILE;
+
+
+if (shardCount) {
+    const jasmine = require("jasmine-core");
+    const fs = require("fs");
+ 
+    // Sharding protocol:
+    // Tell Bazel that this test runner supports sharding by updating the last modified date of the
+    // magic file
+    let fd = fs.openSync(shardStatusFile, "w");
+    fs.closeSync(fd);
+    
+
+    let getTotalSpecsDefined;
+
+    const SuiteBuilder = jasmine.SuiteBuilder;
+    jasmine.SuiteBuilder = ($j) => {
+        return class ProxySuiteBuilder extends SuiteBuilder($j) {
+            constructor(options) {
+                super(options);
+                getTotalSpecsDefined = () => this.totalSpecsDefined;
+            }
+        };
+    }
+
+    const Order = jasmine.Order;
+    jasmine.Order = ($j) => {
+        const RandomOrder = Order($j);
```

Comment:
Why "RandomOrder" instead of "DefaultOrder"? Do we know it's random?

Reply:
Good point. DefaultOrder is reasonable